.github/workflows/ci.yml

```diff
@@ -11,175 +11,229 @@ concurrency:
+          toolchain: stable
+          override: true
 
-    - uses: Swatinem/rust-cache@b894d59a8d236e2979b247b80dac8d053ab340dd # v2.1.0
+      - uses: Swatinem/rust-cache@b894d59a8d236e2979b247b80dac8d053ab340dd # v2.1.0
 
-    - name: Run ipfs-kad example
-      run: RUST_LOG=libp2p_swarm=debug,libp2p_kad=trace,libp2p_tcp=debug cargo run --example ipfs-kad --features full
+      - name: Run ipfs-kad example
+        run: RUST_LOG=libp2p_swarm=debug,libp2p_kad=trace,libp2p_tcp=debug cargo run --example ipfs-kad --features full
 
   rustfmt:
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v3
+      - uses: actions/checkout@v3
 
-    - uses: actions-rs/toolchain@16499b5e05bf2e26879000db0c1d13f7e13fa3af # v1.0.7
-      with:
-        profile: minimal
-        toolchain: stable
-        override: true
-        components: rustfmt
+      - uses: actions-rs/toolchain@16499b5e05bf2e26879000db0c1d13f7e13fa3af # v1.0.7
+        with:
+          profile: minimal
+          toolchain: stable
+          override: true
+          components: rustfmt
 
-    - name: Check formatting
-      run: cargo fmt -- --check
+      - name: Check formatting
+        run: cargo fmt -- --check
 
   manifest_lint:
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v3
-
-    - uses: actions-rs/toolchain@16499b5e05bf2e26879000db0c1d13f7e13fa3af # v1.0.7
-      with:
-        profile: minimal
-        toolchain: stable
-        override: true
-
-    - name: Ensure `full` feature contains all features
-      run: |
-        ALL_FEATURES=$(cargo metadata --format-version=1 --no-deps | jq -r '.packages[] | select(.name == "libp2p") | .features | keys | map(select(. != "full")) | sort | join(" ")')
-        FULL_FEATURE=$(cargo metadata --format-version=1 --no-deps | jq -r '.packages[] | select(.name == "libp2p") | .features["full"] | sort | join(" ")')
-        
-        test "$ALL_FEATURES = $FULL_FEATURE"
-
-        echo "$ALL_FEATURES";
-        echo "$FULL_FEATURE";
-        
-        test "$ALL_FEATURES" = "$FULL_FEATURE"
-
-  gather_crates_for_semver_checks:
+      - uses: actions/checkout@v3
+
+      - uses: actions-rs/toolchain@16499b5e05bf2e26879000db0c1d13f7e13fa3af # v1.0.7
+        with:
+          profile: minimal
+          toolchain: stable
+          override: true
+
+      - name: Ensure `full` feature contains all features
+        run: |
+          ALL_FEATURES=$(cargo metadata --format-version=1 --no-deps | jq -r '.packages[] | select(.name == "libp2p") | .features | keys | map(select(. != "full")) | sort | join(" ")')
+          FULL_FEATURE=$(cargo metadata --format-version=1 --no-deps | jq -r '.packages[] | select(.name == "libp2p") | .features["full"] | sort | join(" ")')
+          
+          test "$ALL_FEATURES = $FULL_FEATURE"
+          
+          echo "$ALL_FEATURES";
+          echo "$FULL_FEATURE";
+          
+          test "$ALL_FEATURES" = "$FULL_FEATURE"
+
+  gather_published_crates:
```

Comment:
Since everything depends on it, my suggestion would be to place this job at the top. Not a requirement but it'd make it easier to read.

Reply:
I usually follow a different philosophy:

The most important jobs should be at the top. Assuming sufficiently descriptive names, you shouldn't need the implementation of a ~function~ job to continue reading and understanding what is happening.

If you do the reverse  - putting all "leaf"[^1] jobs at the top - you end up reading a lot of unrelated snippets  before getting to the parts that actually matter.

[^1]: As in, the leafs in a dependency-tree.